### PR TITLE
Config Fixes: `strict` parsing and deprecation warning

### PR DESCRIPTION
**Pull request by ziadsawalha**

### CODE REVIEW

File: simpl/config.py [hunk 1]
```diff
@@ -656,6 +672,14 @@ def parse_passthru_args(argv):
             return argv[:dashdash], argv[dashdash + 1:]
         return argv, []
 
+    def set_keyring_value(self, key, value, namespace=None):
```
Comment (samstav) [inline simpl/config.py]: Cool. Can you leave an example on how/why to use this?

Comment (ziadsawalha) [inline simpl/config.py]: Oops - this was meant to go i a separate PR.

File: simpl/config.py [hunk 2]
```diff
@@ -259,10 +259,7 @@ def add_argument(self, parser, permissive=False, **override_kwargs):
                 kwargs['help'] = "%s (or set %s)" % (kwargs['help'],
                                                      kwargs['env'])
             if permissive:
-                try:
-                    required = kwargs.pop('required', None)
-                except KeyError:
-                    pass
+                required = kwargs.pop('required', None)
```
Comment (samstav) [inline simpl/config.py]: Good catch- that the try/except is not needed.

**Comment by samstav** [comment]
LGTM, you finished with this one?

**Comment by ziadsawalha** [comment]
Yes. Ready.
